### PR TITLE
Adds Windows touch support, improves mouse button handling

### PR DESCRIPTION
**Pull request by offalynne**

### CODE REVIEW

File: scripts/input_mouse_check/input_mouse_check.gml [hunk 1]
```diff
@@ -8,36 +8,31 @@ function input_mouse_check(_binding)
         return (_binding == mb_none);
     }
     
-    //Extended touchpad handling
-    if (os_type == os_windows)
+    if ((_binding >= mb_right) && (_binding <= mb_side2))
```
Comment (JujuAdams) [inline scripts/input_mouse_check/input_mouse_check.gml]: hmm

File: scripts/input_mouse_check_pressed/input_mouse_check_pressed.gml [hunk 2]
```diff
@@ -2,47 +2,37 @@
...
-    if (!global.__input_mouse_allowed_on_platform || global.__input_window_focus_block_mouse || global.__input_cleared)
+{
+    if (!global.__input_mouse_allowed_on_platform || global.__input_window_focus_block_mouse)
     {
         return (_binding == mb_none);
     }
     
-    //Extended mouse button handling
-    if (os_type == os_windows)
+    if ((_binding >= mb_right) && (_binding <= mb_side2))
```
Comment (JujuAdams) [inline scripts/input_mouse_check_pressed/input_mouse_check_pressed.gml]: hmmmm

File: scripts/input_mouse_check_released/input_mouse_check_released.gml [hunk 3]
```diff
@@ -3,23 +3,36 @@
...
 {    
-    if (!global.__input_mouse_allowed_on_platform || global.__input_cleared)
+    if (!global.__input_mouse_allowed_on_platform || global.__input_window_focus_block_mouse)
     {
-        return false;
+        return (_binding == mb_none);
     }
     
-    if (__INPUT_TOUCH_SUPPORT)
+    if ((_binding >= mb_right) && (_binding <= mb_side2))
```
Comment (JujuAdams) [inline scripts/input_mouse_check_released/input_mouse_check_released.gml]: hmmmmmmmm

**Comment by JujuAdams** [comment]
Also, what version of GM should be used to test this PR?

**Comment by offalynne** [comment]
Runtime 2022.9 adds Windows multi-touch, changes are back compatible

**Comment by offalynne** [comment]
> _hmmmmmmmm_

as-you-wish-princess

File: scripts/input_mouse_check_released/input_mouse_check_released.gml (hunk 3 above)
Comment (offalynne) [reply scripts/input_mouse_check_released/input_mouse_check_released.gml]: fixed (?)

**Comment by JujuAdams** [comment]
Cool cool, I'll use LTS to test in that case.

**Comment by JujuAdams** [comment]
Unable to rebind a verb to `mb_left` when on a Windows device and using a mouse, This appears to be down to https://github.com/JujuAdams/Input/blob/master/scripts/__input_class_source/__input_class_source.gml#L259  If I remove this line then I am able to rebind, though the binding name is then `touchscreen press`. Unsure what the solution here is? Should `INPUT_TOUCH_POINTER_ALLOWED` be involved in this check (and also the binding name check)?

**Pull request merged**